src/app/onboarding/availability/page.tsx

```diff
@@ -1,24 +1,93 @@
 'use client';
 
-import Button from '@/components/Button';
-import COLORS from '@/styles/colors';
+import { useContext, useState, useEffect } from 'react';
 import { H1 } from '@/styles/text';
 import { OnboardingContext } from '@/utils/OnboardingProvider';
-import { useContext } from 'react';
+import TextInput from '@/components/TextInput';
+import { isValidDate } from '@/utils/helpers';
+import DateInput from '@/components/DateInput';
+import { Profile } from '@/types/schema';
+import TextAreaInput from '@/components/TextAreaInput';
 
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const [hours, setHours] = useState('');
+  const [startDate, setStartDate] = useState('');
+  const [periods, setPeriods] = useState('');
+
+  const getHoursErrorText = () => {
+    if (hours !== '' && Number.isNaN(parseInt(hours, 10))) {
+      return 'Must be a number';
+    }
+    return '';
+  };
+
+  const getErrorText = () => {
+    if (startDate !== '' && !isValidDate(startDate)) {
+      return 'Must select a current or future date';
+    }
+    return '';
+  };
+
+  useEffect(() => {
+    console.log(onboarding?.profile.start_date);
+
+    setHours(
+      onboarding?.profile.hours_per_month
+        ? onboarding?.profile.hours_per_month.toString()
+        : '',
+    );
+    setStartDate(onboarding?.profile.start_date || '');
+    setPeriods(onboarding?.profile.availability_description || '');
+  }, []);
```

Comment:
I would recommend adding `onboarding` to the dependency array here so this can update every time onboarding changes

Reply:
at least while testing, if i include `onboarding`, for some reason it causes an  "depth exceeded" error -- as if `onboarding` keeps changing (infinite loop moment) 
would be nice to figure out what is happening there!